substratevm/src/com.oracle.svm.truffle/src/com/oracle/svm/truffle/TruffleBaseFeature.java

```diff
@@ -625,6 +630,112 @@ private static void registerReflectionAccessesForRuntimeValidation(Class<?> stor
     }
 }
 
+@TargetClass(className = "com.oracle.truffle.api.staticobject.StaticProperty", onlyWith = TruffleBaseFeature.IsEnabled.class)
+final class Target_com_oracle_truffle_api_staticobject_StaticProperty {
+
+    @Alias @RecomputeFieldValue(kind = Kind.Custom, declClass = Target_com_oracle_truffle_api_staticobject_StaticProperty.OffsetTransformer.class) //
+    int offset;
+
+    public static final class OffsetTransformer implements RecomputeFieldValue.CustomFieldValueTransformer {
+        /*
+         * We have to use reflection to access private members instead of aliasing them
+         * in the substitution class since substitutions are present only at runtime
+         */
+        private static final Method staticPropertyGetInternalKind;
+
+        static {
+            // Checkstyle: stop
+            staticPropertyGetInternalKind = ReflectionUtil.lookupMethod(StaticProperty.class, "getInternalKindName");
+            // Checkstyle: resume
+        }
+
+        @Override
+        public Object transform(MetaAccessProvider metaAccess, ResolvedJavaField original, ResolvedJavaField annotated,
+                        Object receiver, Object originalValue) {
+            int offset = (int) originalValue;
+            StaticProperty receiverStaticProperty = (StaticProperty) receiver;
+
+            String internalKindName;
+            try {
+                // Checkstyle: stop
+                internalKindName = (String) staticPropertyGetInternalKind.invoke(receiverStaticProperty);
+                // Checkstyle: resume
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                throw VMError.shouldNotReachHere(e);
+            }
+
+            int baseOffset;
+            int indexScale;
+            JavaKind javaKind;
+            switch (internalKindName) {
```

Comment:
Yes, a simple `if` statement comparing the `internalKindName` with `Object` and using `Object` offsets in such case and `byte` offsets in all other cases will also work fine. I updated the re-computation based on the assumption that SVM will use the same base offsets for all primitives. The only reason why initially I used a switch statement was maintenance in case this ever changes.

Reply:
There's no assumption there: the new code is correct, _especially_ if SVM starts using different base offsets: primitives are all stored in a byte array so we must use the offset and scales of byte arrays for both HotSpot and SVM.